src/software/ai/navigator/obstacle/obstacle_factory.h

```diff
@@ -52,8 +69,56 @@ class ObstacleFactory
      *
      * @return A list of obstacles representing the given team
      */
-    std::vector<Obstacle> getVelocityObstaclesFromTeam(const Team &team);
+    std::vector<ObstaclePtr> createVelocityObstaclesFromTeam(const Team &team);
+
+    /**
+     * Create circle obstacle around robot with additional radius scaling
+     *
+     * @param robot_position robot position
+     *
+     * @return obstacle around the robot
+     */
+    ObstaclePtr createRobotObstacle(const Point &robot_position);
+
+    /**
+     * Create circle obstacle around ball
+     *
+     * @param ball_position ball position to make obstacle around
+     *
+     * @return obstacle around the ball
+     */
+    ObstaclePtr createBallObstacle(const Point &ball_position);
+
+    /**
+     * Create rectangle-shaped obstacle
+     *
+     * @param rectangle Rectangle to make obstacle with
+     *
+     * @return rectangular obstacle
+     */
+    ObstaclePtr createObstacleFromRectangle(const Rectangle &rectangle);
```

Comment:
Is a rectangle a subclass of Polygon? We already have `createObstacle(const Polygon &polygon)` do we need this function?

Reply:
createObstacle(const Polygon &polygon is a private helper function that doesn't add a robot radius around the polygon

Reply:
ah gotcha, github folded it weird for me didn't see the `private`